### DISCUSSION TITLE
How to statefully fuzz without invariants?

### DISCUSSION THREAD

**Discussion by antoniopgs** [foundry-rs/foundry#7140] [Help]
Explanation here:
https://ethereum.stackexchange.com/questions/159581/how-to-statefully-fuzz-in-foundry-without-invariants

**Comment by grandizzy** [comment]
You don't need all the invariant handlers / contracts, just write a regular test and add params to fuzz, see book https://book.getfoundry.sh/forge/fuzz-testing

**Reply by antoniopgs** [reply]
But this will only fuzz it statelessly, not statefully.

**Reply by grandizzy** [reply]
ah, sorry, missed that, you're looking for reverts only?

**Reply by antoniopgs** [reply]
Yes.

**Reply by grandizzy** [reply]
Remove `invariant` from your test name

```Solidity
function fuzz_callHandlerFunctions() external {
    }
```

and make sure `fail_on_revert` is true ( default)
